src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -349,11 +349,17 @@ internal Task ExtractRelativeToDirectoryAsync(string destinationDirectoryPath, b
                     throw new InvalidDataException(SR.TarEntryHardLinkOrSymlinkLinkNameEmpty);
                 }
 
-                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath, LinkName);
+                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath,
+                    Path.IsPathFullyQualified(LinkName) ? LinkName : Path.Join(Path.GetDirectoryName(fileDestinationPath), LinkName));
```

Comment:
Is it possible to move this inside `GetSanitizedFullPath`?

Reply:
`GetSanitizedFullPath` does not have the 'is it a link' context.